lib/Dialect/TTIR/IR/TTIROps.cpp

```diff
@@ -126,6 +126,36 @@ ::mlir::LogicalResult mlir::tt::ttir::TransposeOp::verify() {
   return success();
 }
 
+::mlir::LogicalResult mlir::tt::ttir::ReshapeOp::verify() {
+  ::mlir::RankedTensorType outputType = getOutput().getType();
+  auto shape = getShape();
+  int64_t shape_size = static_cast<int64_t>(shape.size());
+
+  if (shape_size == 0) {
+    return emitOpError("Shape attribute must be non-empty");
+  }
+
+  if (shape_size > 5) {
+    return emitOpError("Shape attribute must have at most 5 elements");
```

Comment:
Why is this the case?

Reply:
From pybind they have two "definitions" of `reshape`
1. Reshape which takes tensor and shape argument.
2. Reshape which takes tensor and array which can't have more then 5 elements.

Now 2nd reshape is implemented using 1st reshape but with addition that it allows one dim to be -1. I'm not sure where is 5 comming from I'm waiting for metal folks to send me all constraints...